src/dawn/Optimizer/StencilInstantiation.cpp

```diff
@@ -995,10 +995,10 @@ class StencilDescStatementMapper : public ASTVisitor {
 //===------------------------------------------------------------------------------------------===//
 
 StencilInstantiation::StencilInstantiation(OptimizerContext* context,
-                                           const sir::Stencil* SIRStencil, const SIR* SIR)
+                                           const std::shared_ptr<sir::Stencil> SIRStencil,
+                                           const std::shared_ptr<SIR> SIR)
     : context_(context), SIRStencil_(SIRStencil), SIR_(SIR) {
   DAWN_LOG(INFO) << "Intializing StencilInstantiation of `" << SIRStencil->Name << "`";
-  DAWN_ASSERT_MSG(SIRStencil, "Stencil does not exist");
```

Comment:
Do we still want to check that the shared-ptr does not point to null
`DAWN_ASSERT_MSG(SIRStencil.get(), "Stencil does not exist");`

Reply:
done